src/extension/providers/debug_config_provider.ts

```diff
@@ -522,7 +522,7 @@ export class DebugConfigProvider implements DebugConfigurationProvider {
 		debugConfig.additionalProjectPaths = debugConfig.additionalProjectPaths || vs.workspace.workspaceFolders?.map((wf) => fsPath(wf.uri));
 		debugConfig.args = debugConfig.args || [];
 		debugConfig.vmAdditionalArgs = debugConfig.vmAdditionalArgs || conf.vmAdditionalArgs;
-		debugConfig.toolArgs = await this.buildToolArgs(debugType, debugConfig, conf);
+		debugConfig.toolArgs = await this.buildToolArgs(debugType, debugConfig, conf, deviceManager?.portFromLocalExtension);
```

Comment:
I don't think you need to pass this all the way down, the implementation of `buildFlutterToolArgs` is inside this class, so it can probably read `this.deviceManager?.portFromLocalExtension` directly?